### DISCUSSION TITLE
Use regular font for a term in a term list by default

### DISCUSSION THREAD

**Discussion by m-kru** [typst/typst#4292] [General]
### Description

Currently bold font by default is used for a term in a term list. This is usually the desired behavior. However, sometimes user might want to format the term in a different fashion. For example, take a look at this term I have created using groff.

![image](https://github.com/typst/typst/assets/26227737/8a33ffdc-ba82-456e-a8bd-5cb5e9fc5979)

The term is first bold monospaced, then only monospaced. With typst this is impossible (or cumbersome?) to recreate because all text in the term is bold by default.  If the term text was rendered using the regular font, the user would be able to easily format the term in a desired way. Maybe the term should have an extra `bold: bool` parameter allowing to turn on/off the bold of the term?

### Use Case

Straightforward formatting of the term in a desired way. The term is formatted exactly the same as the regular text is.

**Comment by laurmaedje** [comment]
You can write a show rule for `terms` and format it however you want. It's a bit more complicated than ideal if you just want to get rid of the bold, but adding a `bold` parameter would be very specific.

For reference, the internal terms show rule currently looks something like this:
```typ
#show terms: it => pad(left: it.indent + it.hanging-indent, stack(
  ..it.children.map(item => {
    h(-it.hanging-indent)
    item.term
    it.separator
    item.description
  }),
  spacing: if it.tight { 
    par.leading 
  } else if it.spacing == auto {
    1.2em // block.below doesn't work yet
  } else { 
    it.spacing 
  },
))
```
This is with all the bells and whistles. You could probably make do with less depending on how your document is configured.

Ideally, we'd have a simpler way to reformat parts of a show rule, but figure this is out is complicated.

**Reply by m-kru** [reply]
I have been using typst for 2 weeks. What I have appreciated the most so far is the fact that is seems like it can be used by both, power users, and "dumb" users. I am the dumb user. I try to avoid programming in my document as much as possible. I prefer to choose simple mechanisms even if they lead to repetitions. For example, I separate listings in my document with lines, something like this:
````
#line(length: 100%)
```C
Some C code.
```
#line(length: 100%)
````
I know that I could write a function that would insert these lines automatically, but I don't want to. Why? Because I don't want to learn typst internals programming. What is more, I do not want to introduce unnecessary abstraction or indirection layers.

In my humble opinion, it would be nice if typst provided as much power for dumb users as possible. The bold for terms is a great example here. Imagine that the term is formatted as a regular text. If you wanted to bold the terms you would simply write.

```typ
/ *term1*:
  Term 1 description.

/ *term2*:
  Term 2 descritpion.
```

This is super simple, even the dumb user knows how to bold text with `*`. This requires two extra characters per term, but this is conceptually super simple.

Now, let's analyze what is the default behavior of typst. The typst now tries to outsmart the dumb user by bolding the terms by default. This seems nice at first because in most cases this is what you want. However, if you want to change the default formatting of the term, you are facing a task that is way to hard for the dumb user. I don't understand a damn thing from your code snippet. And to be honest, at the moment I don't even want to understand it.

Typst is great and I want to use it. However, I want to stay away from programming as much as possible. Typst clearly has 2 types of users, dumb and power. I don't know whether you try to target both types, but if you want, then I think the default behavior should be adjusted for the dumb users. Dumb users often have to write more characters, but they accept it and they value simplicity.

I am not sure whether you understand my point. As a dumb user, I would like to have the same possibilities as the power user. However, I don't want to program inside `.typ` files. On the other hand, I accept writing more verbose markup.

**Reply by laurmaedje** [reply]
I do understand your point. It's just a hard balance to strike somewhat good defaults with easy resets for everything. This case is one where it's comparatively hard to get rid of the defaults.

**Reply by m-kru** [reply]
I can see two very simple solutions for this particular issue:

1. Add an extra parameter to the `terms`, something like `no-default-term-format: bool` or `user-term-format: bool`.
2. Use the regular font for the term by default.

Sometimes the best solution is just to do nothing, and give full control to the user. With solution number 2, both dumb and power users have full control. What is more, there is less code to maintain. This would of course break backward compatibility, but the major revision is still 0.

**Reply by laurmaedje** [reply]
Another parameter is an easy fix for this specific issue, but then there will be another element that is hard to reset and over time we will have tons of such parameters instead of a general solution.

As for the second option, it feels like giving up and just having bad looking defaults.

I get your point, I really do, but I think this problem deserves a more general solution and it just isn't here yet.

**Reply by m-kru** [reply]
> As for the second option, it feels like giving up and just having bad looking defaults.

To be honest I don't understand argument about the bad looking. groff uses such an approach and it looks quite nice in my opinion.
![image](https://github.com/typst/typst/assets/26227737/6f4b7b4a-5420-4fbf-8afc-8c007cc727a3)

Groff takes a little bit different approach. You just specify the offset from the beginning of the line where you want to start the description. The start of the descriptions are always aligned.  If the description can't be started in the same line as the term (because there is no space), then the description is started in the next line. 

This:
1. Looks nice.
2. Is conceptually simple.
3. Allows for straightforward, custom formatting of the term.

**Comment by m-kru** [comment]
If someone is facing the same issue. You can come around it by defining a custom command for your term description paragraph.

```typ
#let term-desc(desc) = par(first-line-indent: 1em, hanging-indent: 1em)[#desc]
```
Then you can use it like this:
```typ
*`log`*`(x, b real) integer|real` \
#cmds.term-desc[
The `log` function returns the logarithm of `x` to the base `b`.
]

*`u2`*`(x, w integer) integer` \
#cmds.term-desc[
The `u2` function returns two’s complement representation of `x` as an integer assuming width `w`.
For example, `u2(-1, 8)` returns `255`.
]
```

What gives the following output:
![image](https://github.com/typst/typst/assets/26227737/378177a8-9016-4a82-85d2-105d64203a7b)

**Reply by m-kru** [reply]
The even simpler way is to use the `#pad` function. 
```typ
*`log10`*`(x real) integer|real` \
#pad(left: 1em)[
The `log10` function returns the decimal logarithm of `x`.
]
```

**Reply by laurmaedje** [reply]
Another very manual solution is to reverse the effect of the bold, like this
```typ
#let unbold(it) = strong(delta: -300, it)
/ #unbold[term]: item
```
or like this:
```typ
#let unbold(it) = text(weight: "thin", it)
/ #unbold[term]: item
```

**Comment by hurzl** [comment]
Ideally it should be possible to have a parameter which takes all the parameters to be passed to the term's text, sth. like
```typ 
#set terms(term-text: (weight: "bold", size: 12pt))
```
but then I miss smallcaps, why?
And I can't define a fixed width for the term to match a hanging-indent of the text.
If I want to make my own term definition, can I redefine the "/ x:" shortcut?

**Reply by laurmaedje** [reply]
A more likely solution is that we'll introduce a new subelement, so that you can write e.g. `#show terms.term: set text(weight: "bold", size: 12pt)`.

> If I want to make my own term definition, can I redefine the "/ x:" shortcut?

Yeah, but you need to build it from scratch with a `show terms` rule as shown [here](https://github.com/typst/typst/discussions/4292#discussioncomment-9595259).

**Comment by Geronymos** [comment]
If you want to style the term or definition you can use 

```typst
#show terms.item: it => {
  show it.term.text: it => text(fill: red, it);
  show it.description.text: it => emph(it);
  it
}
```

You can remove the bold font for terms like this:

```typst

#show terms: it => {
  show strong: it => it.body;
  it
}

```

<img width="179" height="197" alt="grafik" src="https://github.com/user-attachments/assets/ac7b74fc-2445-4e52-bdf5-312d1f6c8c5b" />
